### PR TITLE
DisableSyntax.noDefaultArgs add corner case with token position

### PR DESCRIPTION
**Pull request by MasseGuillaume**

### CODE REVIEW

File: scalafix-tests/input/src/main/scala/test/DisableSyntaxMoreRules.scala [hunk 1]
```diff
@@ -53,6 +53,12 @@ Default args makes it hard to use methods as functions.
     def foobarfoo(x: Int = 42) = 6 // assert: DisableSyntax.defaultArgs
   }
 
+  class Bar { type Foo = Int; def foo = 42 }
+  def foo(a: { type Foo = Int; def foo: Foo } = new Bar): Int = a.foo /* assert: DisableSyntax.defaultArgs
```
Comment (MasseGuillaume) [inline scalafix-tests/input/src/main/scala/test/DisableSyntaxMoreRules.scala]:
```scala
def foo(a: { type Foo = Int; def foo: Foo } = new Bar): Int = a.foo
                                              ^ (param.default)
                                            ^ (token)
```

File: scalafix-core/shared/src/main/scala/scalafix/internal/rule/DisableSyntax.scala [hunk 2]
```diff
@@ -81,7 +81,10 @@ final case class DisableSyntax(
                 params <- d.paramss
                 param <- params
                 default <- {
-                  param.tokens.collect { case eqs: Token.Equals => eqs }
+                  param.tokens.reverse.find {
```
Comment (vovapolu) [inline scalafix-core/shared/src/main/scala/scalafix/internal/rule/DisableSyntax.scala]:
```scala
def foo(a: { type Foo = Int; def foo: Foo } = {val a = new Bar, a}): Int = a.foo
```
:D

File: scalafix-core/shared/src/main/scala/scalafix/internal/rule/DisableSyntax.scala [hunk 3]
```diff
@@ -81,7 +81,10 @@ final case class DisableSyntax(
...
+                    case eqs: Token.Equals => true
+                    case _ => false
+                  }.toList
```
Comment (MasseGuillaume) [inline scalafix-core/shared/src/main/scala/scalafix/internal/rule/DisableSyntax.scala]: or `param.default.toList`

File: scalafix-core/shared/src/main/scala/scalafix/internal/rule/DisableSyntax.scala (hunk 2 above)
Comment (MasseGuillaume) [reply scalafix-core/shared/src/main/scala/scalafix/internal/rule/DisableSyntax.scala]: dam

Comment (olafurpg) [reply scalafix-core/shared/src/main/scala/scalafix/internal/rule/DisableSyntax.scala]: Just use `param.default.pos`

Comment (MasseGuillaume) [reply scalafix-core/shared/src/main/scala/scalafix/internal/rule/DisableSyntax.scala]: aww guys that extra 2 chars :p

File: scalafix-tests/input/src/main/scala/test/DisableSyntaxMoreRules.scala (hunk 1 above)
Comment (olafurpg) [reply scalafix-tests/input/src/main/scala/test/DisableSyntaxMoreRules.scala]: There is no benefit to pointing the caret at `=` over `new`, and with range positions it will be way clearer if `new Bar` is highlighted instead of only `=`
